equinix/resource_metal_device.go

```diff
@@ -442,6 +458,42 @@ func shouldReinstall(_ context.Context, d *schema.ResourceDiff, meta interface{}
 	return !reinstall_config["enabled"].(bool)
 }
 
+func shouldReinstallOrAllowChanges(attribute string) customdiff.ResourceConditionFunc {
+	return func(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+		reinstall := shouldReinstall(ctx, d, meta)
```

Comment:
I got the wrong impression reading this at first. Perhaps renaming `reinstall` to `forceNow` would help with legibility.

Reply:
For the first cut I aligned with the (incorrect) `shouldReinstall` function name; I've since renamed both functions to have names that more accurately describe what they're doing.